### PR TITLE
Bugfix/206 ensure expected day is selected

### PR DESCRIPTION
**Pull request by ckunki**
Adjusted some time-outs
added wait until in ProjectRow
added TenaciousChecker
enhanced class Tester for manual live system tests
CHANGELOG

### CODE REVIEW

**Comment by ckunki** [comment]
Hi @kaklakariada, your opinion on this PR would be very valuable to me.

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        private int iterations = 0;
+        private final Driver driver;
+        // private final TenaciousChecker checker;
+        private Duration duration = Duration.ZERO;
+
+        public Testee(int failures)
+        {
+            this.failures = failures;
+            this.driver = new Driver(null, null);
+            // checker = new TenaciousChecker(condition());
```
Comment (kaklakariada) [inline plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java:52]:
```suggestion
```

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        assertThat(testee.getIterations()).isEqualTo(4);
+        assertThat(testee.getDuration().toMillis()).isGreaterThan(29);
+    }
+
+    private static class Testee
+    {
+        private final int failures;
+        private int iterations = 0;
+        private final Driver driver;
+        // private final TenaciousChecker checker;
```
Comment (kaklakariada) [inline plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java:45]:
```suggestion
```

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                {
+                    return ++iterations > failures;
+                }
+            };
+        }
+
+        public boolean check(Duration timeout)
+        {
+            final long start = System.currentTimeMillis();
+            // final boolean result = checker.check(timeout);
```
Comment (kaklakariada) [inline plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java:80]:
```suggestion
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Thank you very much for your pull request, i have only some minor findings.

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java (hunk 3 above)
Comment (ckunki) [reply plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java]: Ooops - thank you very much!
Sonar complains about < 80% coverage on new code. 
But in this case I would tend to ignore that.

Comment (kaklakariada) [reply plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/web/TenaciousCheckTest.java]: Yes, unless you want to create integration tests using pmsmart ;)

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**